<?xml version="1.0"?>
<launch>
    <arg name="init_instruction" default=""/>
    <arg name="use_spacemouse" default="false"/>

    <rosparam file="$(find hsr_data_collection)/config/hsr_data_collection_config.yaml" command="load"/>

    <node name="rosbag_manager" pkg="hsr_data_collection" type="rosbag_manager.py" output="screen">
    </node>

    <node name="status_led_manager" pkg="hsr_data_collection" type="status_led_manager.py" output="screen">
    </node>

    <node name="dualshock_teleop_interface" pkg="hsr_data_collection" type="dualshock_interface.py" output="screen">
        <param name="init_instruction" value="$(arg init_instruction)"/>
        <param name="use_spacemouse" value="$(arg use_spacemouse)"/>
    </node>

    <group if="$(arg use_spacemouse)">
        <include file="$(find spacenav_node)/launch/classic.launch">
        </include>
        <node pkg="hsr_data_collection" name="spacemouse_interface" type="spacemouse_interface.py" output="screen">
        </node>
    </group>

</launch>